### ISSUE TITLE
ocsigen-widgets cannot be installed from opam

### ISSUE THREAD

**Issue by chinwn** [ocsigen/ojwidgets#8]
Following the tutorial, I tried to pin install ocsigen-widget but got the following
circular dependency messages:
# using Ocaml 4.02.0 or 4.02.3

ocsigen-widgets needs to be installed.
The following dependencies couldn't be met:
- ocsigen-widgets -> ojquery -> js_of_ocaml < 2.4
  Your request can't be satisfied:
- js_of_ocaml<2.4 is not available because your system doesn't comply with ocaml-version <= "4.01.0".
# going back to  4.01.0, I got the following error instead.

ocsigen-widgets needs to be installed.
[ERROR] ocsigen-widgets.dev is not available because your system doesn't comply
        with ocaml-version >= "4.02".
[NOTE] Pinning command successful, but your installed packages may be out of
       sync.

**Comment by vasilisp** [comment]
Please also pin `ojquery`:

```
opam pin add ojquery https://github.com/ocsigen/ojquery.git
```

We need to fix the tutorial. Did you find the instructions in ["writing a client/server Eliom application"](https://ocsigen.org/tuto/dev/manual/application), or in some other page?

**Comment by chinwn** [comment]
Yes, this  pin add ocsigen-widgets  command was in the above tutorial.

I followed your suggestion to pin opam pin ojquery, and made more progress.
However, I am faced with the following error when I tried:
  opam pin add ocsigen-widgets https://github.com/ocsigen/ocsigen-widgets.git 

```
> ### stderr ###
> # [...]
> # Use Bytes.set instead.
> # File "src/tools/ow_upload.eliom", line 54, characters 28-40:
> # Warning 3: deprecated: String.set
> # Use Bytes.set instead.
> # File "src/tools/ow_upload.eliom", line 55, characters 28-40:
> # Warning 3: deprecated: String.set
> # Use Bytes.set instead.
> # File "src/widgets/ow_spinner.eliom", line 37, characters 7-33:
> # Error: Unbound value Eliom_config.get_debugmode
> # make: *** [_server/src/widgets/ow_spinner.type_mli] Error 2
```

=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install ocsigen-widgets dev
The following changes have been performed
  ∗  install imagemagick 0.34-1
[NOTE] Pinning command successful, but your installed packages may be out of
       sync.
